app/presenters/two_factor_auth_code/generic_delivery_mode_presenter.rb

```diff
@@ -0,0 +1,42 @@
+module TwoFactorAuthCode
+  class GenericDeliveryModePresenter
+    include ActionView::Helpers::TagHelper
+    include ActionView::Helpers::TranslationHelper
+    include Rails.application.routes.url_helpers
+
+    def initialize(data_model)
+      data_model.each do |key, value|
+        instance_variable_set("@#{key}", value)
+        self.class.send(:attr_reader, key.to_sym)
```

Comment:
The only issue is the some presenters are going to be instantiated with different variables. For instance, the authentication app (totp) presenter more or less only needs a `current_user`, it doesn't care about (or receive) any of the others.